lib/matplotlib/scale.py

```diff
@@ -280,8 +280,11 @@ class LogTransform(Transform):
 
     def __init__(self, base, nonpos='clip'):
         Transform.__init__(self)
+        if base <= 0 or base == 1:
+            raise ValueError('The log base cannot be <= 0 or == 1')
```

Comment:
What happened before?  Did we just get a ValueError later?

Reply:
The error checking was in LogScale (which calls into LogTransform), but that meant that you could construct invalid LogTransforms which only errored out at use time.  With the change you can't construct invalid LogTransforms *or* LogScales anymore.